homeassistant/components/swiss_public_transport/config_flow.py

```diff
@@ -59,6 +60,9 @@ async def async_step_user(
                 _LOGGER.exception("Unknown error")
                 errors["base"] = "unknown"
             else:
+                await self.async_set_unique_id(
+                    f"{user_input[CONF_START]} {user_input[CONF_DESTINATION]}"
+                )
```

Comment:
Shouldn't we abort if there's already an existing config entry with this unique id? Move this up to the top and use our other helper instead of checking for matching config entry data.

https://github.com/home-assistant/core/blob/8e83356ccb1c94cbbc06ca96da11d520387d923b/homeassistant/config_entries.py#L1567

Reply:
@MartinHjelmare There is already a unique entry check at the beginning. The unique_id will only be based on the config entry values and does not have any other useful way of generating something unique. Therefore unique config entry == unique_id should hold.

Please inform me, if I misunderstood that!

https://github.com/miaucl/ha/blob/0257cd8bbed5ddc33d3f795ac70b580c8d70684e/homeassistant/components/swiss_public_transport/config_flow.py#L42

Reply:
We can replace that line with the method I linked above if we move setting the unique id up top when handling the user input.

Reply:
If I understand correctly, you prefer using `abort_if_unique_id_configured` to `async_abort_entries_match`?

Reply:
Yes. We set a unique_id, then we should use that abort method.